### DISCUSSION TITLE
[Enhancement] Vector icons (SVG/XAML)

### DISCUSSION THREAD

**Discussion by LarinLive** [dotnet/maui#136] [General]
## Summary
Please add full support of vector icons in both formats: SVG and XAML. Icons shoud be able to load from files, from resource, or from XAML directly. There should be an abiltiy to place these icons in any control where an image drawing is possible.

It would be nice to have an abiltity to apply color transforms to vector icons by styles: theming, tinting, decolorizing, and so on.

**Comment by charlesroddie** [comment]
Vector very important. SVG is the default choice.

What do you mean by XAML as a vector format?

**Comment by LarinLive** [comment]
> What do you mean by XAML as a vector format?

I have some icon sets with following XAML icon files:
![image](https://user-images.githubusercontent.com/18666703/83602650-a79f0b00-a57b-11ea-8250-317a7df0742a.png)

**Comment by LarinLive** [comment]
I am not insist on XAML icons. If there is a support of SVG, I will convert them to that format.

**Reply by J-Swift** [reply]
How do you convert them? I have a project that converts SVG to several alternative formats (PDF, android xml drawable) and would consider adding the XAML conversion as well if its able to be automated.

https://github.com/J-Swift/Svg2Vector

**Reply by symbiogenesis** [reply]
Add it!

**Comment by MitchBomcanhao** [comment]
You can draw paths via xaml in Xamarin forms right now. I've only briefly played with it and with some tinkering had something that worked, but it wasn't just copying the xaml I had lying around, I had to tweak it a bit. 
Maybe there are easier ways to get it to work, but it definitely is possible to specify the vectorial shapes in xaml and render them in Xamarin forms right now.
